WordPress/gutenberg
# Renders `DataForm` component only when data has been fetched

## What?


This PR renders `DataForm` component only when data has been fetched. 
## Why?


We experienced runtime crashes when certain fields expected defined data but received a boolean value (`false`). This occurred because `getEditedEntityRecord` was still fetching data and returned `false` as a fallback.

Instead, according to type definition, fields expect to receive the `Item`:

https://github.com/WordPress/gutenberg/blob/e4df49ffc87d7f2076d4ea7e0e8bc6698bcc60ff/packages/dataviews/src/types.ts#L549-L555

To avoid runtime crashes, `DataForm` component needs to run only when data has been fetched.

## How?


## Testing Instructions






1. Enable the "Quick Edit in DataViews" experiment
2. Enable the 2024 or 2025 theme and go to **Appearance > Editor > Pages**
3. Change the view in the top right to **Table** and open the right side bar.
4. Select an item.
5. On trunk, add a `console.log(data)` [on this line](https://github.com/WordPress/gutenberg/blob/e4df49ffc87d7f2076d4ea7e0e8bc6698bcc60ff/packages/fields/src/fields/featured-image/featured-image-edit.tsx#L24-L25)
6. Refresh the page.
7. Check the logs and ensure that you see "false" value in the first lines in the logs..
8. Switch to this branch.
9. Repeat the test.
10.  Check the logs and ensure that you don't see "false" value in the first lines in the logs, but only an object.





| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/7d3365d7-ad8a-405a-b634-ff8925bbd9cd" /> | <video src="https://github.com/user-attachments/assets/812a917d-172f-47a7-87a5-1f77a4213f73" />|
